### PR TITLE
Будажапова Екатерина. Задача 1. Вариант 23. Частота символов в cтроке

### PR DESCRIPTION
**Pull request by Save-Yourself**

### CODE REVIEW

File: tasks/mpi/budazhapova_e_count_freq_character/func_tests/count_freq_of_ch_func_tests_mpi.cpp [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+
+  srand(time(nullptr));
+  for (int i = 0; i < length; i++) {
+    result[i] = charset[rand() % charset.length()];
+  }
+  return result;
+}
+}  // namespace
+}  // namespace budazhapova_e_count_freq_chart_mpi
+
```
Comment (aobolensk) [inline tasks/mpi/budazhapova_e_count_freq_character/func_tests/count_freq_of_ch_func_tests_mpi.cpp:29]: Please add simple fixed tests (without random) and some corner cases tests (empty strings, no char encounters, string consists only with these chars, etc.)

File: tasks/seq/budazhapova_e_count_freq_character/func_tests/count_freq_of_ch_func_tests_seq.cpp [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+  task_data_seq->outputs_count.emplace_back(out.size());
+
+  // Create Task
+  budazhapova_e_count_freq_chart_seq::TestTaskSequential test_task_sequential(task_data_seq);
+  ASSERT_EQ(test_task_sequential.Validation(), true);
+  test_task_sequential.PreProcessing();
+  test_task_sequential.Run();
+  test_task_sequential.PostProcessing();
+  ASSERT_EQ(out[0], 1);
+}
```
Comment (aobolensk) [inline tasks/seq/budazhapova_e_count_freq_character/func_tests/count_freq_of_ch_func_tests_seq.cpp:69]: Please, add random tests here as well

**Review by aobolensk — CHANGES_REQUESTED** [review]

File: tasks/mpi/budazhapova_e_count_freq_character/func_tests/count_freq_of_ch_func_tests_mpi.cpp (hunk 1 above)
Comment (Save-Yourself) [reply tasks/mpi/budazhapova_e_count_freq_character/func_tests/count_freq_of_ch_func_tests_mpi.cpp:29]: Thanks, it's done

File: tasks/seq/budazhapova_e_count_freq_character/func_tests/count_freq_of_ch_func_tests_seq.cpp (hunk 2 above)
Comment (Save-Yourself) [reply tasks/seq/budazhapova_e_count_freq_character/func_tests/count_freq_of_ch_func_tests_seq.cpp:108]: Thanks, it's done

File: tasks/seq/budazhapova_e_count_freq_character/src/count_freq_of_ch_src_seq.cpp [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+  res_ = 0;
+  return true;
+}
+
+bool budazhapova_e_count_freq_chart_seq::TestTaskSequential::ValidationImpl() {
+  return task_data->inputs_count[0] == 1;
+}
+
+bool budazhapova_e_count_freq_chart_seq::TestTaskSequential::RunImpl() {
+  for (unsigned long i = 0; i < input_.length(); i++) {
```
[the remark this answers is not in the document]
Comment (Save-Yourself) [reply tasks/seq/budazhapova_e_count_freq_character/src/count_freq_of_ch_src_seq.cpp]: thanks, it's done

**Review by aobolensk — APPROVED** [review]
